error_handling/README.md

```diff
@@ -0,0 +1,223 @@
+# Hello World - Attesting a Function Call
+
+This example shows you a useful pattern for reporting errors during the 
+execution of functions in the Blocky Attestation Service (Blocky AS).
+
+Before starting this example, make sure you are familiar with the
+[Hello World - Attesting a Function Call](../hello_world_attest_fn_call/README.md)
+example.
+
+In this example, you'll learn how to:
+
+- Use the [result pattern](https://en.wikipedia.org/wiki/Result_type) in your
+functions
+- Return structured data from your functions
+- Report errors from your functions
+  
+## Setup
+
+- Install the Blocky AS CLI by following the
+  [setup instructions](https://blocky-docs.redocly.app/attestation-service/setup)
+  in the Blocky AS documentation.
+- Make sure you also have
+  [Docker](https://www.docker.com/) and [jq](https://jqlang.org/) installed on
+  your system.
+
+## Quick Start
+
+To run this example, call:
+
+```bash
+make run-success
+```
+
+You will see the following output extracted from a Blocky AS response:
+
+```json
+{
+  "Success": true,
+  "Error": "",
+  "Value": {
+    "number": 42
+  }
+}
+```
+
+## Walkthrough
+
+### Step 1: Create success and error functions
+
+Before we get to error handling, let's create a function that successfully
+returns structured data. In [`main.go`](./main.go), we define the
+`successFunc` function:
+
+```go
+func successFunc(inputPtr, secretPtr uint64) uint64 {
+	type Output struct {
+		Number int `json:"number"`
+	}
+	output := Output{42}
```

Comment:
Note that while I do sometimes do this in quick examples, it is often recommended to be explicit.  For example:

```
output := Output{ Number: 42 }
```
instead of 
```
output := Output{42}
```

Reply:
Updating